src/reporter-object-factory.ts

```diff
@@ -227,21 +235,65 @@ export default function reporterObjectFactory (
         async reportTestDone (name, testRunInfo): Promise<void> {
             if (rejectReport) return;
 
-            const { screenshots, videos, errs, durationMs, testId, browsers, skipped, unstable } = testRunInfo;
+            const { screenshots, videos, errs, durationMs, testId, browsers, skipped, unstable, fixture } = testRunInfo;
+            const { layoutTestingEnabled, screenshotsDir, destinationDir, comparerBaseDir }               = layoutTestingSettings;
 
-            const testRunToScreenshotsMap: Record<string, string[]> = {};
-            const testRunToVideosMap: Record<string, string[]>      = {};
-            const testRunToErrorsMap: Record<string, TestError>     = {};
+            const testRunToScreenshotsMap: Record<string, ScreenshotMapItem[]> = {};
+
+            const testRunToVideosMap: Record<string, string[]>  = {};
+            const testRunToErrorsMap: Record<string, TestError> = {};
+
+            const testBrowserRuns               = browserToRunsMap[testId];
+            const shouldUploadLayoutTestingData = getShouldUploadLayoutTestingData(layoutTestingEnabled, browsers);
 
             if (!noScreenshotUpload) {
```

Comment:
[minor] Don't you think that `noScreenshotUpload` and `layoutTestingEnabled` options should be independent?

Reply:
So, when `noScreenshotUpload === true` and `layoutTestingEnabled === true` we should upload screenshots (including the current one) only for layout testing failures, right?

It may be useful (although maybe not quite intuitive for users)...

However, in this instance I'll better do this in the next PR where I get a clear (and new) condition that tells that it's a layout testing failure. I introduced this condition to support "new baseline" and "mismatches sizes" scenarios.

If I decouple `noScreenshotUpload` and `layoutTestingEnabled` in this PR, I will still have to redo it in a different way in the next PR.

Reply:
ok